libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1647,8 +1647,22 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		else if (functionName->name() == "suicide" && functionType->kind() == FunctionType::Kind::Selfdestruct)
 			m_errorReporter.typeError(_functionCall.location(), "\"suicide\" has been deprecated in favour of \"selfdestruct\"");
 	}
-	if (!m_insideEmitStatement && functionType->kind() == FunctionType::Kind::Event)
-		m_errorReporter.typeError(_functionCall.location(), "Event invocations have to be prefixed by \"emit\".");
+
+	// Declaring it here since it's used here and below again
+	bool const abiEncodeV2 = m_scope->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::ABIEncoderV2);
+
+	if (functionType->kind() == FunctionType::Kind::Event)
+	{
+		if (!m_insideEmitStatement)
+			m_errorReporter.typeError(_functionCall.location(), "Event invocations have to be prefixed by \"emit\".");
```

Comment:
Isn't this handled in the syntax checker already?

Reply:
This was already there, I just split the `if` conditions to reuse one of them.

Reply:
Oh right, somehow missed it.